In [1]:
%matplotlib notebook
import numpy as np
import matplotlib.pyplot as plt
import pandas
import cv2
from sklearn import svm
from sklearn.metrics import mean_squared_error
from sklearn.neighbors import NearestNeighbors
from speed_regression import training_data as td
from speed_regression import regression
from speed_regression import regression_cascade

nano_to_sec = 1e09

fig_size = plt.rcParams['figure.figsize']
fig_size[0] = 10
fig_size[1] = 8

In [2]:
option = td.TrainingDataOption()

In [13]:
# load testing data
class_map = {'handheld':0, 'leg':1, 'bag':2, 'body':3}
train_feature, train_label, train_responses, _ = regression_cascade.load_datalist('../../data2/list_small.txt',
                                                                              option, class_map)
test_feature, test_label, test_responses, _ = regression_cascade.load_datalist('../../data2/list_small2.txt',
                                                                              option, class_map)
train_responses = train_responses[:, [0, 2]]
test_responses = test_responses[:, [0, 2]]
feature_classes = [test_feature[test_label==cls].astype(np.float32)
                   for cls in model.class_map.values()]
response_classes = [test_responses[test_label==cls].astype(np.float32)
                    for cls in model.class_map.values()]

Loading dataset ../../data2/hang_handheld_normal1/processed/data.csv, type: handheld
Smoothing target with sigma:  30.0
Smoothing the signal by gaussian filter: sigma =  2.0
Loading dataset ../../data2/hang_handheld_speed1/processed/data.csv, type: handheld
Smoothing target with sigma:  30.0
Smoothing the signal by gaussian filter: sigma =  2.0
Loading dataset ../../data2/hang_handheld_side1/processed/data.csv, type: handheld
Smoothing target with sigma:  30.0
Smoothing the signal by gaussian filter: sigma =  2.0
Loading dataset ../../data2/hang_handheld_spin1/processed/data.csv, type: handheld
Smoothing target with sigma:  30.0
Smoothing the signal by gaussian filter: sigma =  2.0
Loading dataset ../../data2/hang_leg_normal1/processed/data.csv, type: leg
Smoothing target with sigma:  30.0
Smoothing the signal by gaussian filter: sigma =  2.0
Loading dataset ../../data2/hang_leg_speed1/processed/data.csv, type: leg
Smoothing target with sigma:  30.0
Smoothing the signal by gaussian fil

NameError: name 'model' is not defined

In [8]:
# Train knn
knn = NearestNeighbors(n_neighbors=3, algorithm='ball_tree', n_jobs=6).fit(train_feature)

(42201, 3) (42201, 3)


In [24]:
indices = knn.kneighbors(test_feature, return_distance=False)

In [25]:
predicted = np.empty(test_responses.shape)
for i in range(test_responses.shape[0]):
    predicted[i] = np.average(test_responses[indices[i, :].ravel()], axis=0)
for i in range(2):
    mse = mean_squared_error(y_true=test_responses[:, i], y_pred=predicted[:, i])
    print('mse for channel {}:, {}'.format(i, mse))

mse for channel 0:, 0.22058399478815982
mse for channel 1:, 0.15514312680554818


# KNN with ball tree (k=3)
* Training accuracy: (0.006702, 0.004230)
* Testing accuracy: (0.220584, 0.155143)